### PR TITLE
browser/gui: Handle CSS special backgrounds

### PR DESCRIPTION
**Pull request by robinlinden**
If the html or body element has a non-transparent background set, that should be used as the canvas background and not just as the element background.

This is currently done in hacky way outside of //render right now as doing correctly, not painting over things outside of the web page, without hacks in //render will require //gfx functionality not in place yet.

Before, html background ending where the html element ends:
Before

After, html background filling the entire canvas:
after

### CODE REVIEW

File: browser/gui/app.cpp
Comment (GrayHatter) [inline browser/gui/app.cpp:494]: isn't it time to write a tree selector api?

**Review by GrayHatter — APPROVED** [review]
LGTM

File: browser/gui/app.cpp
Comment (robinlinden) [reply browser/gui/app.cpp:494]: I have written `dom::nodes_by_xpath`, and I have extended it to work with both the style and layout tree, but all of that stuff is blocked waiting for reviews. Any improvements to that function will automagically work for the trees wrapping dom nodes once I've finished merging the changes to it.

TL;DR: it was time a while ago, and it is done*. :P

**Review by mkiael — APPROVED** [review]

**Pull request merged**
